tests/bluetooth/bsim/mesh/src/test_dfu.c

```diff
@@ -185,10 +248,23 @@ static void target_dfu_transfer_end(struct bt_mesh_dfu_srv *srv, const struct bt
 	ASSERT_TRUE(expect_dfu_xfer_end);
 	ASSERT_TRUE(success);
 
+	if (expected_stop_phase == DFU_SRV_STOP_VERIFY) {
+		k_sem_give(&dfu_ended);
+		return;
+	}
+
 	if (dfu_verify_fail) {
 		bt_mesh_dfu_srv_rejected(srv);
+		if (expected_stop_phase == DFU_SRV_STOP_VERIFY_FAIL) {
```

Comment:
Can we replace `dfu_verify_fail` with `expected_stop_phase == DFU_SRV_STOP_VERIFY_FAIL`?